src/rtctools/data/timeseries.py

```diff
@@ -0,0 +1,86 @@
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import numpy as np
+
+import rtctools.data.csv as csv
+import rtctools.data.pi as pi
+import rtctools.data.rtc as rtc
+
+
+def get_timeseries_from_csv(
+    file: Path, delimiter=","
+) -> Tuple[List[datetime], Dict[str, np.ndarray]]:
+    """Read timeseries from csv
+
+    :param file:        Path to CSV.
+    :param delimiter:   CSV column delimiter.
+
+    :returns: A list of time stamps (datetimes) and dict of values.
+    """
+    data = csv.load(file, delimiter=delimiter, with_time=True)
+    times = data[data.dtype.names[0]]
+    values = {key: np.asarray(data[key], dtype=np.float64) for key in data.dtype.names[1:]}
+    return times, values
+
+
+def get_timeseries_from_pi(
+    file: Path, data_config=rtc.DataConfig, is_binary=False, validate=False
+) -> Tuple[List[datetime], Dict[str, np.ndarray]]:
+    """Read timeseries from Delft-FEWS Published Interface file.
+
+    :param file:        Path to the xml file.
+    :param data_config: A :class:rtc.DataConfig object.
+    :param is_binary:   Indicate if the file is a binary file.
+    :param validate:    Validate the timeseries.
+
+    :returns: A list of time stamps (datetimes) and dict of values.
+    """
+    file = Path(file)
+    timeseries = pi.Timeseries(
+        data_config=data_config,
+        folder=file.parent,
+        basename=file.stem,
+        binary=is_binary,
+        pi_validate_times=validate,
+    )
+    times = timeseries.times
+    values = dict(timeseries.items())
+    return times, values
+
+
+def check_times_are_increasing(times: List[datetime]):
+    """Check that time stamps are increasing."""
+    for i in range(len(times) - 1):
+        if times[i] >= times[i + 1]:
+            raise ValueError("Time stamps must be strictly increasing.")
+
+
+def check_times_are_equidistant(times: List[datetime]):
+    """Check that times are eqeuidistant."""
+    dt = times[1] - times[0]
+    for i in range(len(times) - 1):
+        if times[i + 1] - times[i] != dt:
+            raise ValueError(
+                "Expecting equidistant timeseries, the time step towards "
+                "{} is not the same as the time step(s) before. ".format(times[i + 1])
+            )
+
+
+def fill_nan_in_timeseries(times: List[datetime], values: np.ndarray, interp_args: dict = None):
+    """Fill in missing values in a timeseries using lienar interpolation.
+
+    :param times:   List of datetimes.
+    :param values:  1D array of values with the same length as times.
+    :interp_args:   Dict of arguments passed to numpy.interp.
+
+    :returns:       List of values where nans have been replaced with interpolated values.
+    """
+    if interp_args is None:
+        interp_args = {}
+    times_sec = np.array([(t - times[0]).total_seconds() for t in times])
+    nans = np.isnan(values)
+    result = np.interp(times_sec, times_sec[~nans], values[~nans], **interp_args)
```

Comment:
This leads to an error if the seed is filled with nans. Instead we can already return the function earlier if this is the case 
if nans.all():
        return values  # Return early if all values are NaN
return np.interp(times_sec, times_sec[~nans], values[~nans], **interp_args)

Reply:
Yep, I'll fix it and include this case in the unit test.